### ISSUE TITLE
Argcomplete Framework Extension

### ISSUE THREAD

**Issue by derks** [datafolklabs/cement#299]
Auto-completion for Python/Argparse:

 * https://github.com/kislyuk/argcomplete

```
Tab complete all the things!

Argcomplete provides easy, extensible command line tab completion of arguments 
for your Python script.

It makes two assumptions:

You're using bash or zsh as your shell
You're using argparse to manage your command line arguments/options

Argcomplete is particularly useful if your program has lots of options or subparsers, 
and if your program can dynamically suggest completions for your argument/option 
values (for example, if the user is browsing resources over the network).
```

**Comment by harshadyeola** [comment]
:+1:  Waiting for this feature.

**Comment by derks** [comment]
Added `cement.ext.ext_argcomplete` in `master`, will be included in Cement 2.8. See:

 * http://cement.readthedocs.org/en/latest/api/ext/ext_argcomplete/
